src/SimpleJson.cs

```diff
@@ -561,23 +561,23 @@ public static object DeserializeObject(string json)
 		[SuppressMessage("Microsoft.Design", "CA1007:UseGenericsWhereAppropriate", Justification = "Need to support .NET 2")]
 		public static bool TryDeserializeObject(string json, out object obj)
 		{
-			bool success = true;
-			if (json != null)
-			{
-				char[] charArray = json.ToCharArray();
-				int index = 0;
-				obj = ParseValue(charArray, ref index, ref success);
-			}
-			else
-				obj = null;
-
+            if (json is null)
+            {
+                obj = null;
+                return true;
+            }
+            char[] charArray = json.ToCharArray();
+            int index = 0;
+            bool success = false;
+            obj = ParseValue(charArray, ref index, ref success);
```

Comment:
This block is breaking the JSON loading because it changes `success` to false, which forces the methods ParseValue calls to exit prematurely.